contracts/modules/v2/Claims.sol

```diff
@@ -272,54 +269,46 @@ contract Claims is IClaims, MasterAwareV2 {
       }
       lastClaimSubmissionOnCover[coverId] = ClaimSubmission(uint80(claims.length), true);
     }
-    uint32 coverStart;
-    uint32 coverPeriod;
-    uint8 payoutAsset;
+
+    ICover coverContract = cover();
+    ICover.CoverData memory coverData = cover().coverData(coverId);
+    ICover.CoverSegment memory segment = cover().coverSegments(coverId, segmentId);
+
     {
-      uint96 coverAmount;
-      uint24 productId;
-      ICover coverContract = cover();
-      (
-        productId,
-        payoutAsset,
-        coverAmount,
-        coverStart,
-        coverPeriod,
-        /*uint80 nxmPrice*/
-      ) = coverContract.covers(coverId);
       (
         uint16 productType,
         /*address productAddress*/,
         /*uint payoutAssets*/,
         /* initialPriceRatio */,
         /* capacityReductionRatio */
-      ) = coverContract.products(productId);
+      ) = coverContract.products(coverData.productId);
       (
         /*string descriptionIpfsHash*/,
         uint8 redeemMethod,
         uint16 gracePeriodInDays
       ) = coverContract.productTypes(productType);
       require(redeemMethod == uint8(ICover.RedeemMethod.Claim), "Invalid redeem method");
-      require(requestedAmount <= coverAmount, "Covered amount exceeded");
-      require(coverStart <= block.timestamp, "Cover starts in the future");
+      require(requestedAmount <= segment.amount, "Covered amount exceeded");
+      require(segment.start <= block.timestamp, "Cover starts in the future");
```

Comment:
That's the current assumption. We don't know whether a bug allows it to or if we'll actually want this in the future. It shouln't add too much of a gas cost burden and I'd say it's good to have it just in case.